src/image/transform/mask/minErrorI.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * see http://rsb.info.nih.gov/ij/developer/source/ij/process/AutoThresholder.java.html
+ * The method is present in: An 	iterative implementation of Kittler and Illingworth's Minimum Error
+ * thresholding:Kittler, J & Illingworth, J (1986), "Minimum error thresholding", Pattern Recognition 19: 41-47
+ * @param histogram - the histogram of the image
+ *        total - the number of pixels in the image
+ * @returns {number} - the threshold
+ */
+
+export default function minErrorI(histogram, total) {
+
+    let threshold;
+    let Tprev = -2;
+    let mu, nu, p, q, sigma2, tau2, w0, w1, w2, sqterm, temp;
+    let mean; /* mean gray-level in the image */
+
+    /* Calculate the mean gray-level */
+    mean = 0.0;
+    for (let ih = 0; ih < histogram.length; ih++) {
+        mean += ih * histogram[ih];
+    }
+
+    mean /= total;
+
+    threshold = mean;
+
+    while (threshold !== Tprev) {
+        //Calculate some statistics.
+
+        mu = B(histogram, threshold) / A(histogram, threshold);
+        nu = (B(histogram, histogram.length - 1) - B(histogram, threshold)) / (A(histogram, histogram.length - 1) - A(histogram, threshold));
+        p = A(histogram, threshold) / A(histogram, histogram.length - 1);
+        q = (A(histogram, histogram.length - 1) - A(histogram, threshold)) / A(histogram, histogram.length - 1);
+        sigma2 = C(histogram, threshold) / A(histogram, threshold) - (mu * mu);
+        tau2 = (C(histogram, histogram.length - 1) - C(histogram, threshold)) / (A(histogram, histogram.length - 1) - A(histogram, threshold)) - (nu * nu);
```

Comment:
There are a lot of unneeded calculations here. For example, `A(histogram, threshold)` is calculated 6 times with the same arguments.